tests/test_filer.py

```diff
@@ -199,11 +205,172 @@ class FilerTest_no_env(unittest.TestCase, AssertThrowsMixin):
                            "To enable it, both 'HOST_BASE_PATH' and 'CONTAINER_BASE_PATH' environment variables must be defined."
                          )
+class S3_FilerTest(unittest.TestCase,AssertThrowsMixin):
+
+    def setUp(self):
+        self.mock = mock_s3()
+        self.mock.start()
+        # moto package fails if localhost is given as endpoint_url
+        self.conn = boto3.resource('s3', endpoint_url=None)
+        self.conn.create_bucket(Bucket='tesk')
+        path = "/home/user/filer_test/"
+        url = "http://s3.localhost:9000/tesk/test/bucket_file1.txt"
+        ftype = "FILE"
+        self.trans = S3Transput(path, url, ftype)
+
+    def tearDown(self):
+        self.mock.stop()
+
+    def test_get_names(self):
+        """
+        Check if the bucket name and path is extracted
+        """
+        path = "/home/user/filer_test/"
+        url = "http://mybucket.s3-aws-region.amazonaws.com/folder/file.txt"
+        ftype = "FILE"
+        self.trans = S3Transput(path, url, ftype)
+        self.assertEqual(self.trans.get_names(), ('mybucket', '/folder/file.txt'))
+
+
+    def test_check_if_bucket_exists_success(self):
+        """
+        Check if the bucket exists
+        """
+        client = boto3.resource('s3', endpoint_url=None)
+        self.assertEqual(self.trans.check_if_bucket_exists(client),0)
+
+    def test_check_if_bucket_exists_error(self):
+        """
+        Check if bucket does not exist
+        """
+        path = "/home/user/filer_test/"
+        url = "http://s3.localhost:9000/mybucket/test/bucket_file1.txt"
+        ftype = "FILE"
+        trans = S3Transput(path, url, ftype)
+        client = boto3.resource('s3', endpoint_url=None)
+        self.assertEqual(trans.check_if_bucket_exists(client),1)
+
+    @patch("tesk_core.filer.os.path.exists", return_value=1)
+    def test_extraction_point_file(self,mock_path_exists):
+        """
+        Testing extraction of endpoint url from default file location
+        """
+        read_data = '\n'.join(["[default]", "endpoint_url = http://s3.localhost:9000"])
+        with patch("builtins.open", mock_open(read_data=read_data), create=True) as mock_file:
+            mock_file.return_value.__iter__.return_value = read_data.splitlines()
+            self.assertEqual(self.trans.extract_endpoint(),"http://s3.localhost:9000")
+
+    @patch.dict(os.environ, {"AWS_CONFIG_FILE": "~/.aws/config"})
+    def test_extraction_point_environ(self):
+        """
+        Testing successful extraction of endpoint url read from file path saved on enviornment variable
+        """
+        read_data = '\n'.join(["[default]","endpoint_url = http://s3.localhost:9000"])
+        with patch("builtins.open", mock_open(read_data=read_data),create=True) as mock_file:
+            mock_file.return_value.__iter__.return_value = read_data.splitlines()
+
+            self.assertEqual(self.trans.extract_endpoint(),"http://s3.localhost:9000")
+
```

Comment:
Because you do not check, which file is being actually read, there is no difference in this test and the previous one. It can be any file really that the endpoint is read from and the test does not check that.

Reply:
Added the file check now.